ui-v2/src/components/automations/automations-wizard/actions-step/actions-step.tsx

```diff
@@ -0,0 +1,70 @@
+import { Button } from "@/components/ui/button";
+import { Card } from "@/components/ui/card";
+import { Icon } from "@/components/ui/icons";
+
+import { Form, FormMessage } from "@/components/ui/form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useFieldArray, useForm } from "react-hook-form";
+import { ActionStep } from "./action-step";
+import {
+	ActionsSchema,
+	type ActionsSchema as ActionsSchemaType,
+} from "./action-type-schemas";
+
+type ActionStepsProps = {
+	onPrevious: () => void;
+	onNext: (actionsSchema: ActionsSchemaType) => void;
+};
+
+export const ActionsStep = ({ onPrevious, onNext }: ActionStepsProps) => {
+	const form = useForm<ActionsSchemaType>({
+		resolver: zodResolver(ActionsSchema),
+		defaultValues: {
+			actions: [{ type: undefined }],
+		},
+	});
+
+	const { fields, append, remove } = useFieldArray({
+		control: form.control,
+		name: "actions",
+		shouldUnregister: true,
+		rules: { minLength: 1 },
+	});
+
+	const onSubmit = (values: ActionsSchemaType) => {
+		onNext(values);
+	};
+
+	return (
+		<Card className="p-4 pt-8">
+			<Form {...form}>
+				<form onSubmit={(e) => void form.handleSubmit(onSubmit)(e)}>
+					{fields.map(({ id }, index) => (
+						<ActionStep key={id} index={index} onRemove={() => remove(index)} />
+					))}
+					<Button
+						type="button"
+						variant="outline"
+						onClick={() => append({ type: "cancel-flow-run" })}
```

Comment:
I think it'd be worthwhile to start without an action type selected, but it looks like this suggestion doesn't mesh with the current typing.
```suggestion
						onClick={() => append({ type: undefined })}
```

Reply:
This one I couldn't figure out for a default value, without changing the schema to accept `undefined` (which seems counter intuitive): https://react-hook-form.com/docs/usefieldarray
```
src/components/automations/automations-wizard/actions-step/actions-step.tsx:48:31 - error TS2322: Type 'undefined' is not assignable to type '"cancel-flow-run" | "resume-flow-run" | "suspend-flow-run"'.

48       onClick={() => append({ type: undefined})}
```
I think this might be a limitation with `react-hook-form`

<img width="902" alt="Screenshot 2025-01-13 at 10 28 12 AM" src="https://github.com/user-attachments/assets/327a59bf-6713-46f9-9a18-24a92450dccd" />